presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -731,6 +731,38 @@ public void testPushdownSubfieldsAndJoinReordering()
         }
     }
 
+    @Test
+    public void testBucketFiltering()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE test_bucket_filter1 WITH (bucket_count = 11, bucketed_by = ARRAY['key1']) AS " +
+                    "SELECT orderkey AS key1, comment AS value1 FROM orders");
+            queryRunner.execute("CREATE TABLE test_bucket_filter2 WITH " +
+                    "(bucket_count = 13, partitioned_by = ARRAY[ 'orderstatus' ], bucketed_by = ARRAY[ 'key1', 'key2' ]) AS " +
+                    "SELECT orderkey AS key1, custkey AS key2, comment AS value1, orderstatus FROM orders");
+
+            // Remaining buckets after pruning - 2
```

Comment:
I made the following changes to get these tests to pass:

```
            // Remaining buckets after pruning - 2
            assertPlan(getSession(),
                    "SELECT value1 FROM test_bucket_filter1 WHERE key1 = 101 OR key1 = 60000",
                    anyTree(strictTableScan("test_bucket_filter1", identityMap("key1", "value1"))),
                    plan -> assertTableBucketingLayout(plan, "test_bucket_filter1", 11, ImmutableSet.of(2, 6)));

            // Remaining buckets after pruning - 2
            assertPlan(getSession(),
                    "SELECT value1 FROM test_bucket_filter2 WHERE key1 IN (101, 60000) AND key2 IN (1426, 280)",
                    anyTree(strictTableScan("test_bucket_filter2", identityMap("key1", "key2", "value1"))),
                    plan -> assertTableBucketingLayout(plan, "test_bucket_filter2", 13, ImmutableSet.of(8, 6, 7, 5)));


    private void assertTableBucketingLayout(Plan plan, String tableName, int readBucketCount, Set<Integer> bucketsToKeep)
    {
        TableScanNode tableScan = searchFrom(plan.getRoot())
                .where(node -> isTableScanNode(node, tableName))
                .findOnlyElement();

        assertTrue(tableScan.getTable().getLayout().isPresent());
        HiveTableLayoutHandle layoutHandle = (HiveTableLayoutHandle) tableScan.getTable().getLayout().get();

        assertEquals(layoutHandle.getBucketHandle().get().getReadBucketCount(), readBucketCount);
        assertEquals(layoutHandle.getBucketFilter().get().getBucketsToKeep(), bucketsToKeep);
    }
```

Reply:
It works! Thanks! I changed the layout to display bucketsToKeep which was not displayed earlier.